packages/eds-core-react/src/components/Button/Button.stories.tsx

```diff
@@ -139,46 +178,126 @@ Hierarchy.decorators = [
   ),
 ]
 
-export const FileUpload: Story<ButtonProps> = () => (
-  <label htmlFor="file-upload">
-    <input type="file" id="file-upload" style={{ display: 'none' }} multiple />
-    <Button as="span">Upload file</Button>
-  </label>
-)
-FileUpload.storyName = 'File upload'
-FileUpload.decorators = [
+export const Compact: Story<ButtonProps> = () => {
+  const [compact, setComfortable] = useState<boolean>(true)
+
+  return (
+    <EdsProvider density={compact ? 'compact' : 'comfortable'}>
+      <Checkbox
+        label="Compact"
+        onChange={() => {
+          setComfortable(!compact)
+        }}
+        checked={compact}
+      />
+      <Stack direction="row">
+        <Button>Contained</Button>
+        <Button variant="outlined">Outlined</Button>
+        <Button variant="ghost">Ghost</Button>
+        <Button variant="ghost_icon" aria-label="menu action">
+          <Icon data={menu} title="Ghost icon menu"></Icon>
+        </Button>
+      </Stack>
+    </EdsProvider>
+  )
+}
+Compact.decorators = [
+  (Story) => (
+    <Stack direction="column">
+      <Story />
+    </Stack>
+  ),
+]
+
+export const ProgressButton: Story<ButtonProps> = () => {
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
+
+  const onSubmit = () => {
+    setIsSubmitting(true)
+  }
+
+  return (
+    <>
+      <Button
+        aria-disabled={isSubmitting ? true : false}
+        onClick={!isSubmitting ? onSubmit : undefined}
```

Comment:
Should add `aria-label={isSubmitting ? 'loading data' : null}` whenever there is icon only indicating the state